File: app/src/main/java/com/app/turingrobot/ui/core/BaseActivity.java
package com.app.turingrobot.ui.core;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import com.app.turingrobot.R;
import com.app.turingrobot.core.RobotService;
import com.app.turingrobot.core.App;
import com.app.turingrobot.core.dagger.component.DaggerActivityComponent;
import com.app.turingrobot.core.dagger.module.AModule;
import com.app.turingrobot.helper.SpfHelper;
import com.app.turingrobot.utils.StatusBarUtil;
import com.umeng.analytics.MobclickAgent;
import com.umeng.message.PushAgent;

import javax.inject.Inject;

import io.reactivex.disposables.CompositeDisposable;

/**
 * Created by Alpha on 2016/3/26 20:56.
 */
public class BaseActivity extends AppCompatActivity {

    protected App mApp = App.get();

    @Inject
    protected RobotService apiService;

    @Inject
    protected SpfHelper mSpf;

    protected CompositeDisposable mDispos = new CompositeDisposable();


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        PushAgent.getInstance(this).onAppStart();
        inject();
    }

    private void inject() {
        DaggerActivityComponent.builder()
                .aModule(new AModule())
                .appComponent(mApp.getComponent())
                .build()
                .inject(this);
    }

    @Override
    public void setContentView(View view) {
        super.setContentView(view);
        setStatusBarColor();
    }

    /**
     * 设置状态栏颜色
     */
    protected void setStatusBarColor() {
        StatusBarUtil.setColor(this, ContextCompat.getColor(this, R.color.colorPrimaryDark));
    }

    protected static void start(Context th, Class<?> activity, Bundle bundle) {
        Intent intent = new Intent(th, activity);
        if (bundle != null) {
            intent.putExtras(bundle);
        }
        th.startActivity(intent);
    }

    @Override
    protected void onResume() {
        super.onResume();
        MobclickAgent.onResume(this);
    }

    @Override
    protected void onPause() {
        super.onPause();
        MobclickAgent.onPause(this);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }
}

File: app/src/main/java/com/app/turingrobot/ui/core/BaseActivity.kt
package com.app.turingrobot.ui.core

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v4.content.ContextCompat
import android.support.v7.app.AppCompatActivity
import android.view.View

import com.app.turingrobot.R
import com.app.turingrobot.core.RobotService
import com.app.turingrobot.app.App
import com.app.turingrobot.helper.SpfHelper
import com.app.turingrobot.utils.StatusBarUtil
import com.google.gson.Gson
import com.umeng.analytics.MobclickAgent
import com.umeng.message.PushAgent

import javax.inject.Inject

import io.reactivex.disposables.CompositeDisposable
import okhttp3.OkHttpClient

/**
 * Created by Alpha on 2016/3/26 20:56.
 */
open class BaseActivity : AppCompatActivity() {

    @Inject
    lateinit var apiService: RobotService

    @Inject
    lateinit var mSpf: SpfHelper

    @Inject
    lateinit var okHttp: OkHttpClient

    @Inject
    lateinit var gson: Gson

    val mDisp = CompositeDisposable()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        PushAgent.getInstance(this).onAppStart()
        inject()
    }

    private fun inject() {
        App.component.inject(this)
    }

    override fun setContentView(layoutResID: Int) {
        super.setContentView(layoutResID)
        setStatusBarColor()
    }

    /**
     * 设置状态栏颜色
     */
    protected fun setStatusBarColor() {
        StatusBarUtil.setColor(this, ContextCompat.getColor(this, R.color.colorPrimaryDark))
    }

    override fun onResume() {
        super.onResume()
        MobclickAgent.onResume(this)
    }

    override fun onPause() {
        super.onPause()
        MobclickAgent.onPause(this)
    }

    override fun onDestroy() {
        mDisp.dispose()
        super.onDestroy()
    }

    companion object {

        fun start(th: Context, activity: Class<*>, bundle: Bundle?) {
            val intent = Intent(th, activity)
            if (bundle != null) {
                intent.putExtras(bundle)
            }
            th.startActivity(intent)
        }
    }
}
